docx/header.py

```diff
@@ -8,12 +8,11 @@
     absolute_import, division, print_function, unicode_literals
 )
 
-from .shared import ElementProxy
+from .blkcntnr import BlockItemContainer
 
 
-class Header(ElementProxy):
+class Header(BlockItemContainer):
     """
-    The default page header of a section.
+    Proxy for ``<w:hdr>`` element in this section, having primarily a
+    container role.
     """
-
-    __slots__ = ()
```

Comment:
I used BlockItemContainer in my previous implementation of header and it works great. And its docstring says it's supposed to be used for headers.

It gives you all the `.paragraphs` and `.add_paragraphs` and all that jazz pretty much out of the box.

It's a little bit weird because the class Header needs two separate elements in two separate files to function:
```
<w:headerReference> in document.xml
<w:header> in header1.xml (or whatever)
```